x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/profile/ProfileDomainIntegTests.java

```diff
@@ -249,14 +259,137 @@ public void testGetProfileByAuthenticationWillNotCheckRealmNameForFileOrNativeRe
         assertThat(future1.actionGet().uid(), equalTo(profile1.uid()));
     }
 
+    public void testConcurrentCreationOfNewProfiles() throws InterruptedException {
+        // The profile index may or may not exist
+        if (randomBoolean()) {
+            indexDocument();
+        }
+
+        final String username = randomAlphaOfLengthBetween(5, 12);
+        final RealmDomain realmDomain = randomFrom(
```

Comment:
I randomized the domain using `AuthenticationTests#randomDomain(true)`. 

I intentionally make it possible to have `null` domain because it is perfect legit to run User Profile without any domain configuration and racing should be handled equally.